src/welcomePage.ts

```diff
@@ -37,9 +44,27 @@ export class WelcomePage {
             this.panel.webview.html = html;
             this.panel.onDidDispose(() => {
                 this.panel = undefined;
+                const duration = (new Date().getTime() - startTime.getTime()) / 1000;
+                TelemetryClient.sendEvent(Constants.AzureIoTToolsCloseWelcomePageEvent, { duration: duration.toString() });
+            });
+            this.panel.webview.onDidReceiveMessage((message) => {
+                if (message.href) {
+                    TelemetryClient.sendEvent(Constants.AzureIoTToolsLinkClickEvent, { href: message.href });
+                } else if (message.tab) {
+                    TelemetryClient.sendEvent(Constants.AzureIoTToolsTabClickEvent, { tab: message.tab });
+                }
             });
         } else {
             this.panel.reveal(vscode.ViewColumn.One);
         }
     }
+
+    private needToShow() {
```

Comment:
version is undefined at initial, so this function is always return false?

Reply:
fixed